src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/PeriodicSyncStrategy.java

```diff
@@ -82,6 +137,79 @@ public void clearSyncQueue() {
      */
     @Override
     public int getRemainingCapacity() {
-        return 0;
+        return syncQueue.remainingCapacity();
+    }
+
+    /**
+     * Take and execute items from the sync queue. This is intended to be run in a separate thread.
+     * This will work on all the sync requests in the queue until it's empty.
+     */
+    @SuppressWarnings({"PMD.CompareObjectsWithEquals", "PMD.AvoidCatchingGenericException"})
+    private void syncLoop() {
+        logger.atInfo(SYNC_EVENT_TYPE).log("Start processing sync requests");
+        RetryUtils.RetryConfig retryConfig = this.retryConfig;
+        SyncRequest request = syncQueue.poll();
+        while (request != null) {
```

Comment:
You schedule a thread every X seconds that runs this loop. If this loop takes >= X seconds to process,  you'll have multiple threads running the syncLoop. 

Should you have a check that this isn't already running?

Reply:
Yeah. I agree. I thought that function was surrounded by a lock. But it isn't. Adding a check.